### DISCUSSION TITLE
Deploy Single API from multiple APIs as a microservice

### DISCUSSION THREAD

**Discussion by jayudhandha** [fastapi/fastapi#6315] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# segment/main.py

from fastapi import FastAPI

app = FastAPI()
    
@app.get("/")
def read_root():
    return {"Hello": "Segment"}

# campaign/main.py
from fastapi import FastAPI

app = FastAPI()
    
@app.get("/")
def read_root():
    return {"Hello": "Campaign"}
```

### Description

In below link I saw that we can define routes when we have multiple APIs in Fast API.
https://fastapi.tiangolo.com/tutorial/bigger-applications/

Now my question is, How can i deploy single API from whole project. I want to keep logger, exception & utility classes common and don't want to re-deploy them.

Below if my folder structure. Here in **api** folder, I want to deploy **segment** and **campaign** APIs independently.

<img width="190" alt="FastAPI-Folder Structure" src="https://user-images.githubusercontent.com/5781316/190412130-93938118-5262-4b84-a1a3-54e5e717955f.png">

I'm new in FastAPI so forgive me in advance if i asked silly question :D 

### Operating System

Windows

### FastAPI Version

0.79.0

### Python Version

3.8

**Comment by iudeen** [comment, marked as answer]
You can add one router to other router, if that's what you mean.

`router.include_router(other_router)`

**Comment by jayudhandha** [comment]
What about deployment each API independtly?

**Comment by JarroVGIT** [comment]
You could create containers for each application, and copy the folders you need into it. 

But, in a real world situation, I would make your common modules available in a private pipy repository so you can install them during docker build. And make them projects of them own.

**Comment by mbroton** [comment]
> How can i deploy single API from whole project

I think that in order to provide you with a helpful answer, you need to define what do you mean by "deploy single API".

If you really mean "single API" (process) then it's kinda the opposite of being independent - and can be achieved with routers, as @iudeen suggested.

**Comment by csrgxtu** [comment]
@jayudhandha so you want to deploy `campaign` and `segment` defferently, ie start `campaign` on server-a and `segment` on server-b, right?

1, copy code to server-*
2, if you want to start `campaign` just run following:
```bash
uvicorn api.campaign:app 
```
3, if you want to start `segment` just run following:
```bash
uvicorn api.segment:app
```

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

If that solves the original problem, then you can close this issue @jayudhandha  ✔️

**Comment by jayudhandha** [comment]
Closing this issue as I've used [Router](https://fastapi.tiangolo.com/tutorial/bigger-applications/) for our requirements. 

Thanks,
Jayesh
